install/install.ps1

```diff
@@ -8,87 +8,183 @@ $script:ErrorActionPreference='Stop'
+    $releaseName = "azureauth-${version}-win10-x64"
+    $releaseFile = "${releaseName}.zip"
+    $releaseUrl = "https://github.com/${repo}/releases/download/${version}/$releaseFile"
+
+    $azureauthDirectory = if ([string]::IsNullOrEmpty($Env:AZUREAUTH_INSTALL_DIRECTORY)) {
+        ([System.IO.Path]::Combine($Env:LOCALAPPDATA, "Programs", "AzureAuth"))
     } else {
-        "${currentPath};${latestDirectory}"
+        $Env:AZUREAUTH_INSTALL_DIRECTORY
+    }
+    $extractedDirectory = ([System.IO.Path]::Combine($azureauthDirectory, $releaseName))
+    $versionPath = ([System.IO.Path]::Combine($azureauthDirectory, $version))
+    $zipFile = ([System.IO.Path]::Combine($azureauthDirectory, $releaseFile))
+
+    Write-Verbose "Creating ${azureauthDirectory}"
+    $null = New-Item -ItemType Directory -Force -Path $azureauthDirectory
+
+
+    Write-Verbose "Downloading ${releaseUrl} to ${zipFile}"
+    $client = New-Object System.Net.WebClient
+    $client.DownloadFile($releaseUrl, $zipFile)
+
+    # A running instance of azureauth can cause installation to fail, so we try to kill any running instances first.
+    # We suppress taskkill output here because this is a best effort attempt and we don't want the user to see its output.
+    # Here, Get-Process is used to first determine whether there is an existing azureauth process. If there is, kill the existing process first.
+    $ProcessCheck = Get-Process -Name azureauth -ErrorAction SilentlyContinue -ErrorVariable ProcessError
+    if ($ProcessCheck -ne $null)
+    {
+        Write-Verbose "Stopping any currently running azureauth instances"
+        taskkill /f /im azureauth.exe 2>&1 | Out-Null
+
+        # After killing the process it is still possible for there there to be locks on the files it was using (including
+        # its own DLLs). The OS may take an indeterminate amount of time to clean those up, but so far we've observed 1
+        # second to be enough.
+        Start-Sleep -Seconds 1
     }
-    Set-ItemProperty -Path $registryPath -Name PATH -Value $newPath
+
+    if (Test-Path -Path $extractedDirectory) {
+        Write-Verbose "Removing pre-existing extracted directory at ${extractedDirectory}"
+        Remove-Item -Force -Recurse $extractedDirectory
+    }
+
+    Write-Verbose "Extracting ${zipFile} to ${extractedDirectory}"
+    Add-Type -AssemblyName System.IO.Compression.FileSystem
+    [System.IO.Compression.ZipFile]::ExtractToDirectory($zipFile, $azureauthDirectory)
+
+    Write-Verbose "Removing ${zipFile}"
+    Remove-Item -Force $zipFile
+
+    $registryPath = 'Registry::HKEY_CURRENT_USER\Environment'
+    $currentPath = (Get-ItemProperty -Path $registryPath -Name PATH -ErrorAction SilentlyContinue).Path
+
+    # Permanently add the latest directory to the current user's $PATH (if it's not already there).
+    # Note that this will only take effect when a new terminal is started.
+    if (($currentPath -ne $null) -And !$currentPath.Contains($azureauthDirectory)) {

```

Comment:
A couple of comments from pairing:
1. It will be slightly easier to read the logic, I think, if we invert the if and else cases, so we do not have to invert each boolean contains check. First check if it's on the PATH already, if so build an updated newPath. Else - just append. 
2. We can factor out the actual Path set command, and scope the if branches to just assigning the new path var. 
3. We can move the azureauth get-Command test above the if, so that we can also look for it in the initial PATH testing (i.e. include whatever directory it may be installed in already if it is. Because in this case - if it's already on the PATH appending will not actually add the current version to the PATH since it is already shadowed.